src/services/asset-discovery-service.ts

```diff
@@ -124,33 +151,45 @@ export default class AssetDiscoveryService extends PercyClientService {
   }
 
   async teardown() {
-    await this.closePages()
+    await this.cleanPool()
     await this.closeBrowser()
   }
 
-  private async resourcesForWidth(page: puppeteer.Page, width: number, domSnapshot: string,
-                                  rootResourceUrl: string, enableJavaScript: boolean): Promise<any[]> {
+  private async resourcesForWidth(
+    pool: pool.Pool<puppeteer.Page>,
+    width: number,
+    domSnapshot: string,
+    rootResourceUrl: string,
+    enableJavaScript: boolean,
+  ): Promise<any[]> {
     logger.debug(`discovering assets for width: ${width}`)
 
+    profile('--> assetDiscoveryService.pool.acquire', {url: rootResourceUrl})
```

Comment:
Very very nice. If I read this correctly, we'll get timing information on when we actually do have to wait for a page in the pool to become available.